### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by zhaojie-xu**
add zone filter ; 
add data source vpc

### CODE REVIEW

File: alicloud/data_source_alicloud_vpcs.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+	}
+}
+func dataSourceAlicloudVpcsRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AliyunClient).ecsconn
+
+	cidrBlock, cidrBlockOk := d.GetOk("cidr_block")
+	status, statusOk := d.GetOk("status")
+	nameRegex, nameRegexOk := d.GetOk("name_regex")
+	isDefault, isDefaultOk := d.GetOk("is_default")
+	vswitchId, vswitchIdOk := d.GetOk("vswitch_id")
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_vpcs.go]: Line 98 to 102 can be removed according to the next comment.

File: .idea/dictionaries/danny.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<component name="ProjectDictionaryState">
```
Comment (xiaozhu36) [inline .idea/dictionaries/danny.xml]: Please remove .idea while the next PR.

File: alicloud/data_source_alicloud_vpcs.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+			break
+		}
+
+		args.Pagination = *pagination
+	}
+
+	var filteredVpcsTemp []ecs.VpcSetType
+
+	for _, vpc := range allVpcs {
+		if cidrBlockOk && vpc.CidrBlock != cidrBlock.(string) {
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_vpcs.go]: Line 129 can be simplified by the following code:

if cidrBlock, ok := d.GetOk("cidr_block"); ok && vpc.CidrBlock != cidrBlock.(string) {}

And same as the next lines.

File: alicloud/data_source_alicloud_zones.go [hunk 4]
```diff
@@ -78,14 +78,26 @@ func dataSourceAlicloudZonesRead(d *schema.ResourceData, meta interface{}) error
 
 	var zoneTypes []ecs.ZoneType
 	for _, types := range resp {
+		if insType == "" && len(types.AvailableInstanceTypes.InstanceTypes) == 0 {
+			continue
+		}
+
 		if insType != "" && !constraints(types.AvailableInstanceTypes.InstanceTypes, insType) {
 			continue
 		}
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_zones.go]: Line 81 to 87 can be improved by the following expression:

if len(types.AvailableInstanceTypes.InstanceTypes) == 0 ||
          (insType != "" && !constraints(types.AvailableInstanceTypes.InstanceTypes, insType)) {
         continue
}

Same as the next lines.

File: terraform/examples/alicloud-ecs/main.tf [hunk 5]
```diff
@@ -46,7 +52,8 @@ resource "alicloud_instance" "instance" {
   instance_type = "${data.alicloud_instance_types.instance_type.instance_types.0.id}"
   count = "${var.count}"
   availability_zone = "${var.availability_zones}"
-  security_groups = ["${alicloud_security_group.group.*.id}"]
+  security_groups = [
+    "${alicloud_security_group.group.*.id}"]
```
Comment (xiaozhu36) [inline terraform/examples/alicloud-ecs/main.tf]: Why did line 55 be changed?

Comment (zhaojie-xu) [inline terraform/examples/alicloud-ecs/main.tf]: I'll recover it

File: .idea/dictionaries/danny.xml (hunk 2 above)
Comment (zhaojie-xu) [inline .idea/dictionaries/danny.xml]: OK,I'll be more careful next time.

**Pull request closed without merging**